src/python-frontend/python_converter.cpp

```diff
@@ -641,6 +766,29 @@ bool python_converter::is_constructor_call(const nlohmann::json &json)
   return is_ctor_call;
 }
 
+void python_converter::update_instance_from_self(
+  const std::string &class_name,
+  const std::string &func_name,
+  const std::string &obj_symbol_id)
+{
+  std::string self_id =
+    create_symbol_id() + "@C@" + class_name + "@F@" + func_name + "@self";
+
+  auto self_instance = instance_attr_map.find(self_id);
+  if (self_instance != instance_attr_map.end())
+  {
+    std::set<std::string> &attr_list = instance_attr_map[obj_symbol_id];
+
+    for (const auto &element : self_instance->second)
+    {
+      if (
+        std::find(attr_list.begin(), attr_list.end(), element) ==
```

Comment:
No need for std::find() on a std::set. As a std::set contains unique elements, just .insert() can be used without even checking whether the element existed in the set before. In fact this entire manual loop can be replaced by a call to attr_list.[merge](https://en.cppreference.com/w/cpp/container/set/merge)().

Reply:
Thanks for the comments.
`merge()` removes elements from the source. As they should not be removed, I kept the `insert()` call. In fact, the loop is unnecessary.

Reply:
I see, thanks for the clarification and update.